wrapper.js

```diff
@@ -31,13 +31,41 @@ function setupMethods (soljson) {
     };
   }
 
+  var alloc;
+  if ('_solidity_alloc' in soljson) {
+    alloc = soljson.cwrap('solidity_alloc', 'number', [ 'number' ]);
+  }
+
+  var resize;
+  if ('_solidity_resize' in soljson) {
+    resize = soljson.cwrap('solidity_resize', null, [ 'number', 'number' ]);
+  }
+
+  var reset;
+  if ('_solidity_reset' in soljson) {
+    reset = soljson.cwrap('solidity_reset', null, []);
+  }
+
   var copyToCString = function (str, ptr) {
     var length = soljson.lengthBytesUTF8(str);
     // This is allocating memory using solc's allocator.
-    // Assuming copyToCString is only used in the context of wrapCallback, solc will free these pointers.
-    // See https://github.com/ethereum/solidity/blob/v0.5.13/libsolc/libsolc.h#L37-L40
-    var buffer = soljson._malloc(length + 1);
-    soljson.stringToUTF8(str, buffer, length + 1);
+    //
+    // Before 0.6.0:
+    //   Assuming copyToCString is only used in the context of wrapCallback, solc will free these pointers.
+    //   See https://github.com/ethereum/solidity/blob/v0.5.13/libsolc/libsolc.h#L37-L40
+    //
+    // After 0.6.0:
+    //   The duty is on solc-js to free these pointers. We accomplish that by calling `reset` at the end.
+    var buffer;
+    if (isVersion6) {
+      buffer = alloc(length + 1);
+      soljson.stringToUTF8(str, buffer, length + 1);
+      resize(buffer, length);
```

Comment:
Can you explain why you need the resize? Is it really worth saving a single byte?

Reply:
See https://github.com/ethereum/solidity/pull/7898#issuecomment-563482425

Reply:
It is a c interface, so we should consider this to have a trailing zero. I think it is the responsibility of the C++ part to not take the length property for granted but instead search for the zero terminator.

Reply:
That is listed as an option on the issue. Anyway this change confirms that the interface is working well otherwise (since all tests pass).

We can agree on the final API tomorrow on the main issue.

Reply:
Most of the relevant discussion took place on https://github.com/ethereum/solidity/pull/7898